src/base/MiniStack.hh

```diff
@@ -0,0 +1,66 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file MiniStack.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "Macros.hh"
+#include "Types.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Helper class that provides the functionality of a stack on an underlying
+ * container.
+ */
+template<class T>
+class MiniStack
```

Comment:
I'm ok with `MiniStack` since it's consistent with `std::stack` being a container adaptor, but you're better with the naming so I'll leave it up to you.

Reply:
Hah I'm not better, I just fret about it too much. Let's leave it as is for now.